src/ImageStats/Histogram.h

```diff
@@ -18,25 +18,32 @@ struct HistogramResults {
     int num_bins;
     float bin_width;
     float bin_center;
-    std::vector<int> histogram_bins;
+    std::vector<int> histogram_bins; // histogram bin counts
 };
 
 class Histogram {
-    int _num_bins;
-    float _bin_width;
-    float _min_val;
-    std::vector<int> _hist;
+    float _min_val; // lower bound of the histogram (inclusive)
+    float _max_val; // upper bound of the histogram (inclusive)
+    HistogramResults _histogram; // histogram stats
     const std::vector<float>& _data;
 
 public:
     Histogram(int num_bins, float min_value, float max_value, const std::vector<float>& data);
     Histogram(Histogram& h, tbb::split);
 
     void operator()(const tbb::blocked_range<size_t>& r);
-    void join(Histogram& h); // NOLINT
+    bool join(Histogram& h); // NOLINT
     void setup_bins();
 
-    HistogramResults GetHistogram() const;
+    HistogramResults GetHistogram() const { return _histogram; }
+    float GetMinVal() const { return _min_val; }
+    float GetMaxVal() const { return _max_val; }
+    int GetNbins() const { return _histogram.num_bins; }
+    float GetBinWidth() const { return _histogram.bin_width; }
+    float GetBinCenter()  const { return _histogram.bin_center; }
+    std::vector<int> GetHistogramBins() const { return _histogram.histogram_bins; }
```

Comment:
shouldn't this be a `const std::vector<int>&` return type? do we need to copy all the bins, rather than just getting a ref?

Reply:
Yes, I agree. Do histograms ever change after filling? I am asking this to avoid having changed histogram stats that could be used elsewhere from a previous histogram setup.

Reply:
The only way to change them is to "add" another histogram

Reply:
So the bin contents would change but not the actual bin count or Min/max

Reply:
If you add another histogram, bin counts change so that the histogram results will change as well. I'll add the const and reference, but it's important to keep that in mind for the histogram results usage.

Reply:
Sorry, I meant the number of bins will stay the same, not the bin counts ☺️